store/src/java/com/zimbra/cs/dav/service/method/CalendarMultiget.java

```diff
@@ -48,9 +51,24 @@ public void handle(DavContext ctxt) throws ServiceException, DavException {
 
 		DavResponse resp = ctxt.getDavResponse();
 		ArrayList<String> hrefs = new ArrayList<String>();
-		for (Object obj : query.elements(DavElements.E_HREF))
-			if (obj instanceof Element)
-				hrefs.add(((Element)obj).getText());
+        for (Object obj : query.elements(DavElements.E_HREF)) {
+            if (obj instanceof Element) {
+                String text = ((Element)obj).getText();
+                // just check for '+' and encode
+                // encoding already encoded string will create issues
+                if (text.contains("+")) {
+                    try {
+                        ZimbraLog.dav.debug("going to encode '+' characters in href text=%s", text);
+                        text = text.replace("+", URLEncoder.encode("+", MimeConstants.P_CHARSET_UTF8));
```

Comment:
why do not URL encode the complete text ? 

Remove + sign in all logger statement use  %s instead.

Reply:
```
// just check for '+' and encode
// encoding already encoded string will create issues
```
encoding already encoded string will create issues. please see below example

e.g. 
`+` gets encoded as `%2B`
but if string already contains `%2B`, it will get encoded to `%252B`


all the new added logs are having `%s`